nntrainer/src/pooling2d_layer.cpp

```diff
@@ -111,7 +114,10 @@ sharedConstTensor Pooling2DLayer::backwarding(sharedConstTensor derivative,
   } break;
   case PoolingType::global_max: {
     for (unsigned int i = 0; i < derivative->getDim().getDataLen(); ++i) {
-      out[max_idx[i]] += derivative->getData()[i];
+      float der = derivative->getData()[i] / max_idx_global[i].size();
+      for (unsigned int m = 0; m < max_idx_global[i].size(); m++) {
+        out[max_idx_global[i][m]] += der;
```

Comment:
Looks like it is averaging. is it right implementation?

Reply:
It averages in case there are multiple locations which have the same location and are the max values.
This is the implementation which matches tensorflow.
For global_max, tensorflow requires avg of these max values.
However, for max, tensorflow takes the first max value.
This difference in implementation is shown with int unittests as range (1, 10) produces similar integer numbers for the pooling to operate on.

Reply:
Ohh. I got it. I didn't check there are multiple same max value.